realm/realm-library/src/main/java/io/realm/RealmDictionary.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import org.bson.types.Decimal128;
+import org.bson.types.ObjectId;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.realm.internal.OsMap;
+
+/**
+ * Specialization for {@link RealmMap}s whose keys are strings.
+ * <p>
+ * Similarly to {@link RealmList}s, a RealmDictionary can operate in managed and unmanaged modes. In
+ * managed mode a RealmDictionary persists all its contents inside a Realm whereas in unmanaged mode
+ * it functions like a {@link HashMap}.
+ * <p>
+ * Managed RealmDictionaries can only be created by Realm and will automatically update its content
+ * whenever the underlying Realm is updated. Managed RealmDictionaries can only be accessed using
+ * the getter that points to a RealmDictionary field of a {@link RealmObject}.
+ * <p>
+ * Unmanaged RealmDictionaries can be created by the user and can contain both managed and unmanaged
+ * RealmObjects. This is useful when dealing with JSON deserializers like GSON or other frameworks
+ * that inject values into a class. Unmanaged elements in this list can be added to a Realm using
+ * the {@link Realm#copyToRealm(Iterable, ImportFlag...)} method.
+ *
+ * @param <V> the type of the values stored in this dictionary
+ */
+public class RealmDictionary<V> extends RealmMap<String, V> {
+
+    // ------------------------------------------
+    // Unmanaged constructors
+    // ------------------------------------------
+
+    /**
+     * Instantiates a RealmDictionary in unmanaged mode.
+     */
+    public RealmDictionary() {
+        super();
+    }
+
+    /**
+     * Instantiates a RealmDictionary in unmanaged mode with an initial dictionary.
+     *
+     * @param dictionary initial dictionary
+     */
+    public RealmDictionary(RealmDictionary<V> dictionary) {
```

Comment:
Should this just accept a Map<String, V>? or what is the reason for forcing it to be RealmDictionary?

Reply:
That's right, I don't know why I left it as `RealmDictionary` 😅